ui/components-react/src/components-react/filter-builder/FilterBuilderState.ts

```diff
@@ -0,0 +1,203 @@
+/*---------------------------------------------------------------------------------------------
```

Comment:
When you use react reducers, action declarations and implementations get separated. If we don't need to serialize actions on `FilterBuilderState` (I don't see why we would right now), we can simplify everything with `useState` like this:

```ts
export class FilterBuilderActions {
  constructor(private setState) {}

  public addItem(path, itemType) {
    this.setState((prevState) => {...});
  }
  ...
}

Reply:
I like the approach you suggested. Made changes in https://github.com/iTwin/itwinjs-core/pull/3655/commits/075e8b6d501d5af5184be9f90bdcb75b1c532f33